### PR TITLE
Added missing private headers to cmake

### PR DESCRIPTION
**Pull request by nmoinvaz**
Having all the headers makes it easier to find text in source code when searching using an IDE that cmake supports.

### CODE REVIEW

**Comment by nmoinvaz** [comment]
Merged in latest changes. I don't like to rebase because sometimes it will add my username to all commits it does even though it isn't my checkin.

I also added crc32_p.h.

File: CMakeLists.txt [hunk 1]
```diff
@@ -655,15 +659,23 @@ set(ZLIB_PUBLIC_HDRS
...
     inflate.h
+    inflate_p.h
     inftrees.h
+    match_p.h
+    memcopy.h
     trees.h
+    zbuild.h
+    zendian.h
     zutil.h
 )
```
Comment (Dead2) [inline CMakeLists.txt:681]: Missing fallback_builtins.h and trees_p.h, possibly others, not sure.

Comment (nmoinvaz) [reply CMakeLists.txt:681]: fallback_builtin.h is already added via ZLIB_ARCH_HDRS only for MSVC.

Comment (nmoinvaz) [reply CMakeLists.txt:681]: Added missing trees_p.h

**Comment by nmoinvaz** [comment]
Added trees_p.h and double checked to make sure I didn't miss anything. Not sure why codecov failing. Maybe needs to be retried.

**Comment by Dead2** [comment]
Needs a rebase. Would probably be a good idea if you squashed the commits too.

**Comment by nmoinvaz** [comment]
It was easier to just create a new PR, see #447.

**Pull request closed without merging**